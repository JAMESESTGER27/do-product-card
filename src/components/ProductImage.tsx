import React from "react";
// import { ProductContext } from "./ProductCard";
import styles from "../styles/styles.module.css";
import noImage from "../assets/no-image.jpg";


export interface Props {
  className?: string;
  style?: React.CSSProperties;
  img?:string;
}

export const ProductImage = ({  img,className , style}: Props) => {
  // const {
  //   product: { image },
  // } = useContext(ProductContext);
  return (
    <img
      className={`${styles.productImg} ${className}`}
      src={img ? img : noImage}
      alt="product image"
      style={style}
    />
  );
};
