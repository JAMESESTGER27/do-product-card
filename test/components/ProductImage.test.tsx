import React from 'react';
import renderer from 'react-test-renderer';
import { ProductTitle } from '../../src/components/ProductTitle';
import { ProductCard } from '../../src/components/ProductCard';
import { product2 } from '../data/products';
import { ProductImage } from '../../src/components/ProductImage';

describe('ProductImage', () => {
  test('Debe de mostrar el componente correctamente con la imagen personalizada', () => {
    const wrapper = renderer.create(
      <ProductImage img="https://i.ibb.co/cvpntL1/hats.png" />
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('Debe de mostrar el componente con la imagen producto', () => {
    const wrapper = renderer.create(
      <ProductCard product={product2}>{() => <ProductTitle />}</ProductCard>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
