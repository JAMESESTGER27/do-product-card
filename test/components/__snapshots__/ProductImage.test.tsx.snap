// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage Debe de mostrar el componente con la imagen producto 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee Mug - con imagen
  </span>
</div>
`;

exports[`ProductImage Debe de mostrar el componente correctamente con la imagen personalizada 1`] = `
<img
  alt="product image"
  className="productImg undefined"
  src="https://i.ibb.co/cvpntL1/hats.png"
/>
`;
